src/pages/index.js

```diff
@@ -24,28 +34,134 @@ const enableValidation = (config) => {
+
+      likeMethod(cardId)
+        .then((newCardData) => {
+          console.log("New card data after like/unlike:", newCardData);
+          card.updateLikes(newCardData);
+        })
+        .catch((err) => console.error("Error updating like:", err));
+    },
+    handleDeleteClick: (cardId) => {
+      console.log("Delete clicked", cardId);
+      deleteCardPopup.open();
+
+      const deleteButton = deleteCardPopup._submitButton;
+      if (deleteButton) {
+        deleteButton.disabled = false;
+        deleteButton.classList.remove("modal__button_disabled");
+      }
 
-function createCard(item) {
-  const card = new Card(item, selectors.cardTemplate, handleCardClick);
-  return card.getCardElement();
+      deleteCardPopup.setAction(() => {
+        deleteCardPopup.renderLoading(true);
+        api
+          .deleteCard(cardId)
+          .then(() => {
+            card.deleteCard();
+            deleteCardPopup.close();
+          })
+          .catch((err) => console.error("Error deleting card:", err))
+          .finally(() => {
+            deleteCardPopup.renderLoading(false);
+          });
+      });
+    },
+    userId: userId,
+  });
+  return card.createCard();
+}
+
+function handleProfileFormSubmit(formData) {
+  console.log("Profile form submitted", formData);
+  editProfilePopup.renderLoading(true);
+  api
+    .editProfile(formData.name, formData.description)
+    .then((updatedUser) => {
+      console.log("Profile updated", updatedUser);
+      userInfo.setUserInfo({
+        name: updatedUser.name,
+        description: updatedUser.about,
+        avatar: updatedUser.avatar,
+      });
+      editProfilePopup.close();
+    })
+    .catch((err) => {
+      console.error("Profile update error:", err);
+    })
+    .finally(() => {
+      editProfilePopup.renderLoading(false);
+    });
+}
+
+function handleAvatarFormSubmit(formData) {
+  console.log("Avatar form submitted", formData);
+  avatarEditPopup.renderLoading(true);
+  api
+    .setUserAvatar(formData.avatar)
+    .then((userData) => {
+      console.log("Avatar updated", userData);
+      userInfo.setUserInfo({
+        name: userData.name,
+        description: userData.about,
+        avatar: userData.avatar,
+      });
+      avatarEditPopup.close();
+    })
+    .catch((err) => {
+      console.error("Avatar update error:", err);
+    })
+    .finally(() => {
+      avatarEditPopup.renderLoading(false);
+    });
+}
+
+function handleAddCardFormSubmit(formData) {
+  console.log("Add card form submitted", formData);
+  addCardPopup.renderLoading(true);
+  api
+    .addCard(formData.name, formData.link)
+    .then((newCard) => {
+      console.log("New card added", newCard);
+      const cardElement = createCard(newCard);
+      cardSection.addItem(cardElement);
+      addCardPopup.close();
+      addCardPopup.resetForm(); // Reset the form only after successful submission
+    })
+    .catch((err) => {
+      console.error("Add card error:", err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors